openedx/core/djangoapps/util/user_messages.py

```diff
@@ -181,12 +184,54 @@ def _create_user_message(message):
         return (_create_user_message(message) for message in django_messages if self.get_namespace() in message.tags)
 
 
+# .. toggle_name: GLOBAL_NOTICE_ENABLED
+# .. toggle_implementation: SettingToggle
+# .. toggle_default: False
+# .. toggle_description: When enabled, show the contents of GLOBAL_NOTICE_MESSAGE
+#   as a message on every page. This is intended to be used as a way of
+#   communicating an upcoming or currently active maintenance window or to
+#   warn of known site issues. HTML is not supported for the message content,
+#   only plaintext. Message styling can be controlled with GLOBAL_NOTICE_TYPE,
+#   set to one of INFO, SUCCESS, WARNING, or ERROR (defaulting to INFO). Also
+#   see openedx.core.djangoapps.util.maintenance_banner.add_maintenance_banner
+#   for a variation that only shows a message on specific views.
+# .. toggle_use_cases: open_edx
+# .. toggle_creation_date: 2021-09-08
+GLOBAL_NOTICE_ENABLED = SettingToggle('GLOBAL_NOTICE_ENABLED', default=False)
+
+
 class PageLevelMessages(UserMessageCollection):
     """
     This set of messages appears as top page level messages.
     """
     NAMESPACE = 'page_level_messages'
 
+    @classmethod
+    def user_messages(cls, request):
+        """
+        Returns outstanding user messages, along with any persistent site-wide messages.
+        """
+        msgs = list(super().user_messages(request))
+
+        # Add a global notice message to the returned list, if enabled.
+        try:
+            if GLOBAL_NOTICE_ENABLED.is_enabled():
+                if notice_message := getattr(settings, 'GLOBAL_NOTICE_MESSAGE', None):
+                    notice_type_str = getattr(settings, 'GLOBAL_NOTICE_TYPE', None)
+                    # If an invalid type is given, better to show a
+                    # message with the default type than to fail to
+                    # show it at all.
+                    notice_type = getattr(UserMessageType, notice_type_str, UserMessageType.INFO)
```

Comment:
I _think_ this is the correct way to do a lookup by name on an Enum while allowing a fallback to a default.